odoo/models.py

```diff
@@ -5523,8 +5523,10 @@ def __contains__(self, item):
             return len(item) == 1 and item.id in self._ids
         elif isinstance(item, str):
             return item in self._fields
+        elif isinstance(item, BaseModel):
+            raise TypeError(f"cannot compare different models: '{self._name}()' and '{item._name}()'")
         else:
-            raise TypeError("Mixing apples and oranges: %s in %s" % (item, self))
+            raise TypeError(f"unsupported operand type(s) for in: '{self._name}()' and '{type(item)}'")
```

Comment:
Operation needs to be more visible e.g. quoted or something, as-is the message looks like somebody fucked up and had a stroke.